sky/clouds/service_catalog/data_fetchers/fetch_aws.py

```diff
@@ -62,78 +94,89 @@ def get_spot_pricing_table(region: str) -> pd.DataFrame:
-            'AcceleratorName': acc_name,
-            'AcceleratorCount': acc_count,
-            'vCPUs': get_vcpus(row),
-            'MemoryGiB': get_memory_gib(row),
-        })
-
-    df['Region'] = region
-    df = df.merge(pd.DataFrame(zone_df), how='cross')
-    df = pd.concat([df, df.apply(get_additional_columns, axis='columns')],
-                   axis='columns')
+def get_instance_types_df(region: str) -> Union[str, pd.DataFrame]:
+    try:
+        df, zone_df, pricing_df, spot_pricing_df = ray.get([
+            get_instance_types.remote(region),
+            get_availability_zones.remote(region),
+            get_pricing_table.remote(region),
+            get_spot_pricing_table.remote(region)
+        ])
+        print(f'{region} Processing dataframes')
+
+        def get_acc_info(row) -> Tuple[str, float]:
+            accelerator = None
+            for col, info_key in [('GpuInfo', 'Gpus'),
+                                  ('InferenceAcceleratorInfo', 'Accelerators'),
+                                  ('FpgaInfo', 'Fpgas')]:
+                info = row.get(col)
+                if isinstance(info, dict):
+                    accelerator = info[info_key][0]
+            if accelerator is None:
+                return None, np.nan
+            return accelerator['Name'], accelerator['Count']
+
+        def get_vcpus(row) -> float:
+            if not np.isnan(row['vCPU']):
+                return float(row['vCPU'])
+            return float(row['VCpuInfo']['DefaultVCpus'])
+
+        def get_memory_gib(row) -> float:
+            if isinstance(row['MemoryInfo'], dict):
+                return row['MemoryInfo']['SizeInMiB'] // 1024
+            return int(row['Memory'].split(' GiB')[0])
+
+        def get_additional_columns(row):
+            acc_name, acc_count = get_acc_info(row)
+            # AWS p3dn.24xlarge offers a different V100 GPU.
+            # See https://aws.amazon.com/blogs/compute/optimizing-deep-learning-on-p3-and-p3dn-with-efa/
+            if row['InstanceType'] == 'p3dn.24xlarge':
+                acc_name = 'V100-32GB'
+            if row['InstanceType'] == 'p4de.24xlarge':
+                acc_name = 'A100-80GB'
+                acc_count = 8
+            return pd.Series({
+                'AcceleratorName': acc_name,
+                'AcceleratorCount': acc_count,
+                'vCPUs': get_vcpus(row),
+                'MemoryGiB': get_memory_gib(row),
+            })
+
+        df = df.merge(pricing_df, on=['InstanceType'], how='outer')
+        df['Region'] = region
+        df = df.merge(pd.DataFrame(zone_df), how='cross')
+
+        df = df.merge(spot_pricing_df,
+                      left_on=['InstanceType', 'AvailabilityZone'],
+                      right_index=True,
+                      how='outer')
+
+        df = pd.concat(
+            [df, df.apply(get_additional_columns, axis='columns')],
+            axis='columns')
+        df['GpuInfo'].fillna(df['AcceleratorName'], inplace=True)
```

Comment:
Can you add a few lines of comments that elaborates on what are happening in this part of the code? I find it a bit difficult to follow.